src/Controller/RegistrationController.php

```diff
@@ -26,40 +25,37 @@ public function __construct(EmailVerifier $emailVerifier)
     }
 
     #[Route('/register', name: 'app_register')]
-    public function register(Request $request, UserPasswordHasherInterface $userPasswordHasher, EntityManagerInterface $entityManager): Response
+    public function register(Request $request, UserPasswordHasherInterface $userPasswordHasher, EntityManagerInterface $entityManager): JsonResponse
     {
-        $user = new User();
-        $form = $this->createForm(RegistrationFormType::class, $user);
-        $form->handleRequest($request);
+        try {
+            $user = new User();
+            $resultJson = json_decode($request->getContent(), true);
 
-        if ($form->isSubmitted() && $form->isValid()) {
-            // encode the plain password
+            $user->setEmail($resultJson["username"]);
             $user->setPassword(
                 $userPasswordHasher->hashPassword(
                     $user,
-                    $form->get('plainPassword')->getData()
+                    $resultJson["password"]
                 )
             );
+            $user->setRoles($resultJson["roles"]);
+
+            $userArray = get_object_vars($user);
+            if ( count($userArray) <= 3 ) {
+                return $this->json(['status' => '400', 'message' => 'Erro ao prosseguir com esse cadastro. Preencha todos os campos para prosseguir cadastro'], 200, ['Content-Type'=>'application/json; charset=utf-8']);
+            }
+            $userSearched = $entityManager->getRepository(User::class)->findUserByEmail($user->getEmail());
+            if ( $userSearched > 0 ) {
+                return $this->json(['status' => '400', 'message' => 'Erro ao prosseguir com esse cadastro. O seguinte campo deve ser único: email'], 200, ['Content-Type'=>'application/json; charset=utf-8']);
```

Comment:
Bad practice. You can't give the information that an email is already being used. It helps hacker to know that an email exist then they can try to brute force the password to hack the user.

Reply:
I have a question. How can I send to my front end which error I have to register user? Or can return a list with possibles things cause that error?

Obs: That screen only can access for admin users.